Batch File/Batch File.sublime-syntax

```diff
@@ -48,7 +48,7 @@ variables:
   eoc: (?=\s*{{eoc_char}})
   eoc_char: '[\n|&]'
 
-  label_comment: ':+[^{{label_start}}]'
+  label_comment: ':(?::+|(?!{{label_start}}))'
   label_start: '[^{{metachar}}:+]'
```

Comment:
Can I suggest the following (would require a minor fix in tests):

```suggestion
  label_comment: ':(?:{{label_terminator_chars}}+|(?!{{label_start}}))'
  label_start: '[^{{metachar}}{{label_terminator_chars}}]'
  label_terminator_chars: '[:+]'
```

As `+` (and technically `\s` but probably not relevant here) terminate a label as well ([source](https://www.dostips.com/forum/viewtopic.php?f=3&t=3803)).

Reply:
After a closer look the whole topic seems to be way more complicated than expected.

The following is a valid `label` for instance.

```
:== label ==
```

I guess we should merge #3999 first as this one requires some larger changes, which requiere some of the other changes.